src/Cli/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -41,17 +39,46 @@ public int Execute()
         {
             Initialize();
 
+            if (!TryGetLaunchProfileSettingsIfNeeded(out var launchSettings))
+            {
+                return 1;
+            }
+
             if (ShouldBuild)
             {
+                if (string.Equals("true", launchSettings?.DotNetRunMessages, StringComparison.OrdinalIgnoreCase))
+                {
+                    Reporter.Output.WriteLine(LocalizableStrings.RunCommandBuilding);
+                }
+
                 EnsureProjectIsBuilt();
             }
 
             try
             {
                 ICommand targetCommand = GetTargetCommand();
-                if (!ApplyLaunchProfileSettingsIfNeeded(ref targetCommand))
+                if (launchSettings != null)
                 {
-                    return 1;
+                    if (!string.IsNullOrEmpty(launchSettings.ApplicationUrl))
+                    {
+                        targetCommand.EnvironmentVariable("ASPNETCORE_URLS", launchSettings.ApplicationUrl);
+                    }
+
+                    foreach (var entry in launchSettings.EnvironmentVariables)
+                    {
+                        string value = Environment.ExpandEnvironmentVariables(entry.Value);
+                        if (entry.Key == "ASPNETCORE_HOSTINGSTARTUPASSEMBLIES")
```

Comment:
This is not ideal. SDK is supposed to be neutral. But this is clearly ASPNET specific code. Can it be something generic?

Reply:
Have an abstraction in launchSettings to let a list of entries to be EnvironmentVariables maybe better.